src/ResourceManagement/Logic/Logic.Tests/InMemoryTests/IntegrationAccountSessionInMemoryTests.cs

```diff
@@ -0,0 +1,316 @@
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.NoContent
+            };
+
+            client.Sessions.Delete(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName");
+
+            handler.Request.ValidateAuthorizationHeader();
+            handler.Request.ValidateMethod(HttpMethod.Delete);
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_Get_Exception()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            Assert.Throws<ValidationException>(() => client.Sessions.Get(
+                resourceGroupName: null,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName"));
+            Assert.Throws<ValidationException>(() => client.Sessions.Get(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: null,
+                sessionName: "SessionName"));
+            Assert.Throws<ValidationException>(() => client.Sessions.Get(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: null));
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_Get_OK()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = this.Session
+            };
+
+            var result = client.Sessions.Get(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName");
+
+            handler.Request.ValidateAuthorizationHeader();
+            handler.Request.ValidateMethod(HttpMethod.Get);
+
+            this.ValidateSession(result as IntegrationAccountSession);
+        }
+
+        private void ValidateSessionList(IPage<IntegrationAccountSession> result)
+        {
+            Assert.Equal(1, result.Count());
+            this.ValidateSession(result.First());
+            Assert.Equal(Constants.NextPageLink, result.NextPageLink);
+        }
+
+        private void ValidateSession(IntegrationAccountSession Session)
+        {
+            Assert.Equal("/subscriptions/f34b22a3-2202-4fb1-b040-1332bd928c84/resourceGroups/IntegrationAccountSdkTest/providers/Microsoft.Logic/integrationAccounts/IntegrationAccount1600/sessions/IntegrationAccountAgreement8906-ICN", Session.Id);
```

Comment:
This is not a recorded test. Nevertheless, I am changing the id string validation to a regular expression validation of the format. As this test is a pattern, this will change all entity types validation.